src/SchemaDesign/SchemaDesigner.tsx

```diff
@@ -0,0 +1,416 @@
+                            </div>
+                          )}
+                        </Droppable>
+                      </DragDropContext>
+                      <Button onClick={handleClickOpen}>Add new question</Button>
+                    </Box>
+                  )}
+                </Box>
+              ))}
+              {selectedStep === '' && <Typography>Add a new step to begin designing your schema</Typography>}
+            </Paper>
+          </Box>
+        </Grid>
+        <Grid item xs={12} sm={12} md={6}>
+          <Typography sx={{ mb: 2 }} variant='h4'>
+            Preview
+          </Typography>
+          <Paper
+            sx={{
+              p: 2,
+              width: '100%',
+            }}
+          >
+            {showForm && splitSchema?.steps[stepIndex]?.schema === undefined && (
+              <Typography>Nothing to preview!</Typography>
+            )}
+            {showForm && splitSchema?.steps[stepIndex]?.schema !== undefined && (
+              <SchemaForm
+                schema={splitSchema.steps[stepIndex].schema}
+                formData={splitSchema.steps[stepIndex].state}
+                uiSchema={splitSchema.steps[stepIndex].uiSchema}
+              >
+                {/* eslint-disable-next-line react/jsx-no-useless-fragment */}
+                <></>
+              </SchemaForm>
+            )}
+          </Paper>
+        </Grid>
+      </Grid>
+      <Divider orientation='horizontal' flexItem sx={{ mt: 4, mb: 4 }} />
+      <Box sx={{ textAlign: 'right' }}>
+        <Button variant='contained' onClick={() => setSubmitSchemaDialogOpen(true)} disabled={steps.length === 0}>
+          Submit schema
+        </Button>
+      </Box>
+      <QuestionPicker
+        onSubmit={addQuestion}
+        handleClose={handleQuestionPickerClose}
+        questionPickerOpen={questionPickerOpen}
+      />
+      <Dialog open={newStepDialogOpen} onClose={() => setNewStepDialogOpen(false)}>
+        <DialogTitle>Add new Step</DialogTitle>
+        <Box component='form' onSubmit={addNewStep}>
+          <DialogContent>
+            <DialogContentText>
+              The step name and reference must be unique, if it is left blank it will be generated automatically using
+              the step name. The step name is the title displayed on the form itself.
+            </DialogContentText>
+            <Stack direction='row' spacing={2} sx={{ pt: 2 }}>
+              <TextField
+                label='Step reference'
+                onChange={(event): void => setStepReference(event.target.value)}
+                value={stepReference}
+              />
+              <Stack>
+                <TextField
+                  required
+                  label='Step name'
+                  onChange={(event): void => setStepName(event.target.value)}
+                  value={stepName}
+                />
+                <Typography variant='caption' sx={{ color: '#e35a5a' }}>
```

Comment:
Instead of having a random colour controlled at the component level, can we define this in our theme and give it a descriptive name?

As we've experienced recently, these are the sort of things that can be a pain to resolve when supporting both light and dark modes if they're not controlled at the theme level.

Reply:
We already have` theme.palette.error.main`, so should be fine to use that